src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTest.java

```diff
@@ -232,4 +236,28 @@ public void should_increment_failure_count_when_unable_to_upload_file() throws E
         Envelope envelope = envelopes.get(0);
         assertThat(envelope.getUploadFailureCount()).isEqualTo(1);
     }
+
+    @Test
+    public void should_not_process_again_if_blob_delete_failure()
+        throws Exception {
+        //Given
+        uploadZipToBlobStore(ZIP_FILE_NAME_SUCCESS); //Zip file with metadata and pdfs
+
+        Envelope existingEnvelope = EnvelopeCreator.envelope("A", Status.DELETE_BLOB_FAILURE);
```

Comment:
is it an existing envelope or a new one? surely the UUID will be different?

Reply:
It simulates an existing envelope by creating one before processing the blob. The check is done using the blob container + filename (not the uuid).

Reply:
ok, in that case, imho, it would be more clear to save envelope in db first and then upload a zip to blob storage so that it's processed. (ideally with a comment)

Reply:
That is a good point. Actually the test should check the blob has been deleted and status has been updated. Which is what I've just added.